case_studies/weather_dependent_app_example/appserver/static/javascript/views/app.js

```diff
@@ -0,0 +1,50 @@
+
+/**
+ * This is an example using pure react, with no JSX
+ * If you would like to use JSX, you will need to use Babel to transpile your code
+ * from JSK to JS. You will also need to use a task runner/module bundler to
+ * help build your app before it can be used in the browser.
+ * Some task runners/module bundlers are : gulp, grunt, webpack, and Parcel
+ */
+
+import * as Setup from "./setup_page.js";
+
+define(["react", "splunkjs/splunk"], function(react, splunk_js_sdk){
+  const e = react.createElement;
+
+  class SetupPage extends react.Component {
+    constructor(props) {
+      super(props);
+
+      this.state = {
+        token: ''
+      };
+
+      this.handleChange = this.handleChange.bind(this);
+      this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(event) {
+      this.setState({ ...this.state, [event.target.name]: event.target.value})
+    }
+
+    async handleSubmit(event) {
+      event.preventDefault();
+
+      await Setup.perform(splunk_js_sdk, this.state)
+    }
+
+    render() {
+      return e("div", null, [
+        e("h2", null, "Check to see if dependent app is installed and has the required version. App configuration is not complete without it. If dependency requirements are satisfied, the app configuration will complete."),
```

Comment:
I suggest adjusting this text to explain exactly what the setup is checking for.

Something like:

"This app requires you to install the weather app in order to complete the setup process. If this dependency requirement is satisfied, then app setup completes and you are directed to the app home page. If this dependency requirement is not satisfied, then you are prompted to install the weather app."

Reply:
Fixed.